### ISSUE TITLE
Query Broker crashing when running a custom script

### ISSUE THREAD

**Issue by bpschmitt** [pixie-io/pixie#363]
**Describe the bug**

When running the attached script using [local dev mode](https://docs.pixielabs.ai/tutorials/pxl-scripts/script-dev-environment/), the query broker crashes.

**To Reproduce**

I added a simple function to the `px/nodes` script, return a `test` value and store in a new column called `wavelength_zone`.  

```
def getWavelength(node: str):
    return "test"

def nodes(start_time: str):
    df = px.DataFrame(table='process_stats', start_time=start_time)
    df.node = df.ctx['node_name']
    df['wavelength_zone'] = getWavelength(df.node)
    return df.groupby(['node','wavelength_zone']).agg()
```

**Expected behavior**
I would expect either a syntax error or the script to run successfully.

**Screenshots**
N/A

**Logs**
query broker pod [error.log](https://github.com/pixie-io/pixie/files/7566249/error.log) attached.

**App information (please complete the following information):**
- Pixie version: ??
- K8s cluster version: v1.21.2-eks-55daa9d
- Node Kernel version: 5.4.141-67.229.amzn2.x86_64
- Browser version: Chrome 95.0.4638.69

**Additional context**
N/A

**Comment by bpschmitt** [comment]
Full script attached.

[nodes.pxl.txt](https://github.com/pixie-io/pixie/files/7566282/nodes.pxl.txt)

**Comment by vihangm** [comment]
The trace is from `C.PlannerPlan`, so looks like this is coming from the cpp planner.

**Comment by JamesMBartlett** [comment]
Looks like a bug with how we handle type annotations. I'll fix the bug when I'm back in the office, but for now you can workaround it by removing the type annotation from the definition of `getWavelength`

**Comment by robertprast** [comment]
@JamesMBartlett I looked into this a little bit and found that the issue stems from passing a different type than what is defined in the type annotation. Rather than gracefully handling when `type(obj_var) != type(func_arg_type_annoation)`, the planner will try to force a cast to the type_annoation leading to a fault. 

In the above example, `df.node` is a type of dataframe column but `def getWavelength(node: str)` is trying to force cast it to a `str` leading to the crash.

**Comment by JamesMBartlett** [comment]
Fixed by  1c784be486cd46cc014fbd319255546dc3994d70.
